src/facade/dragonfly_connection.cc

```diff
@@ -1158,11 +1160,19 @@ void Connection::Migrate(util::fb2::ProactorBase* dest) {
   // connections
   CHECK(!cc_->async_dispatch);
   CHECK_EQ(cc_->subscriptions, 0);    // are bound to thread local caches
+  CHECK_EQ(self_.use_count(), 1u);    // references cache our thread and backpressure
   CHECK(!dispatch_fb_.IsJoinable());  // can't move once it started
 
   owner()->Migrate(this, dest);
 }
 
+Connection::WeakRef Connection::Borrow(unsigned thread) {
+  DCHECK(self_);
+  DCHECK_GT(cc_->subscriptions, 0);
+
+  return WeakRef{self_, queue_backpressure_, thread};
+}
```

Comment:
The thread parameter should not be passed, instead it should be socket->proactor()->GetIndex() once the helio change is ready